db/db.py

```
@@ -314,6 +323,22 @@ def read_profile(handle: str) -> Optional[BlueskyProfile]:
         if row is None:
             return None
         
+        # Validate required fields are not NULL
+        if row["handle"] is None:
+            raise ValueError("handle cannot be NULL")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)